src/data/real/nnreal.lean

```diff
@@ -606,6 +612,21 @@ by simp only [←nnreal.coe_lt_coe, nnreal.coe_sub h, nnreal.coe_add, sub_lt_iff
 lemma sub_eq_iff_eq_add {a b c : ℝ≥0} (h : b ≤ a) : a - b = c ↔ a = c + b :=
 by rw [←nnreal.eq_iff, nnreal.coe_sub h, ←nnreal.eq_iff, nnreal.coe_add, sub_eq_iff_eq_add]
 
+lemma mul_sub (a b c : ℝ≥0) : a * (b - c) = a * b - a * c :=
```

Comment:
Are those lemmas that could be generalized to truncated substraction?

Reply:
Could you please point me to the relevant file? It seems that I've missed introduction of another typeclass.

Reply:
There's `algebra.ordered_sub` written by Floris recently. It is not (yet) about interaction with multiplication, but maybe you can make that work without too much of a hassle?
I think you can go by `apply le_antisymm`, then `a * b - a * c ≤ a * (b - c) ↔ a * b ≤ a * (b - c) + a * c`, case-split on `b ≤ c`. And for the other side you can case-split on `a = 0`/`0 < a` then do `a * (b - c) ≤ a * b - a * c ↔ b - c ≤ (a * b - a * c)/a ↔ b ≤ (a * b - a * c)/a + c` then again case-split on `b ≤ c`.

Reply:
And I have the intuition that `a *` can be generalized to a linear (strict?) monotone `f`.

Reply:
We have no typeclass with `group_with_zero α` + `distrib α` but not `add_group α` and I don't want to add one for the sake of these lemmas. And I don't know how to prove this without surjectivity of multiplication by a nonzero element.

Reply:
Okay! Could you then put a TODO close to the lemma and another one in `algebra.ordered_sub`?